### PR TITLE
Add policy properties for aws_iam_role

### LINKED ISSUE #54: Add policy properties for aws_iam_role
<a href="https://github.com/rx294"></img></a> **Issue by [rx294](https://github.com/rx294)**
_Monday Jan 08, 2018 at 22:17 GMT_
_Originally opened as https://github.com/chef/inspec-aws/issues/174_

----

Ref #174 from https://github.com/chef/inspec-aws
Please consider the following features to support cis-aws-foundations-1.18
Enhancements:

```
  describe aws_iam_role(IAM_MASTER_ROLE_NAME) do
    it { should exist }
    its('inline-policies') { should include IAM_MASTER_INLINE_POLICY }
    its('attached-policies') { should include IAM_MASTER_ATTACHED_POLICY }
  end
```

1) List inline policies
`aws iam list-role-policies --role-name <role_name>`
2) List attached policlies
`aws iam list-attached-role-policies --role-name <rolename>`

### PR DESCRIPTION
**Pull request by rmoles**
### Description

Add `scope` and `only_attached` filters for `aws_iam_policies`

### Issues Resolved

Fixes #54 
### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_iam_role.rb [hunk 1]
```diff
@@ -36,6 +37,29 @@ def initialize(opts = {})
         @permissions_boundary_arn  = resp.permissions_boundary.permissions_boundary_arn
       end
     end
+    fetch_role_policies(opts[:role_name])
+    fetch_attached_role_policies(opts[:role_name])
+  end
+
+  def fetch_role_policies(role_name = '')
```
Comment (r-fennell) [inline libraries/aws_iam_role.rb]: `validate_parameters` will fail if `role_name` is `''` so this default can be removed

File: libraries/aws_iam_role.rb [hunk 2]
```diff
@@ -36,6 +37,29 @@ def initialize(opts = {})
...
+    catch_aws_errors do
+      resp = @aws.iam_client.list_role_policies({ role_name: role_name })
+      @inline_policies = resp.policy_names
+    end
+  end
+
+  def fetch_attached_role_policies(role_name = '')
```
Comment (r-fennell) [inline libraries/aws_iam_role.rb]: `validate_parameters` will fail if `role_name` is `''` so this default can be removed

File: libraries/aws_iam_policies.rb [hunk 3]
```diff
@@ -30,7 +30,6 @@ def initialize(opts = {})
     parameters = {}
     parameters[:only_attached] = opts[:only_attached] if opts[:only_attached]
     parameters[:scope]         = opts[:scope]         if opts[:scope]
-    parameters[:path_prefix]   = opts[:path_prefix]   if opts[:path_prefix]
```
Comment (r-fennell) [inline libraries/aws_iam_policies.rb]: Does this removal break any ITs?

Comment (rmoles) [reply libraries/aws_iam_policies.rb]: Nope. I forgot to remove it in a previous PR. Had considered adding it in, but decided against it.

File: libraries/aws_iam_role.rb [hunk 4]
```diff
@@ -35,6 +36,15 @@ def initialize(opts = {})
         @permissions_boundary_type = resp.permissions_boundary.permissions_boundary_type
         @permissions_boundary_arn  = resp.permissions_boundary.permissions_boundary_arn
       end
+      fetch_attached_role_policies(role_name)
+    end
+  end
+
+  def inline_policies
+    return @inline_policies if defined? @inline_policies
```
Comment (r-fennell) [inline libraries/aws_iam_role.rb]: 👍

File: libraries/aws_iam_role.rb [hunk 5]
```diff
@@ -45,4 +55,21 @@ def exists?
...
+      resp = @aws.iam_client.list_attached_role_policies({ role_name: role_name }).attached_policies
+      @attached_policies_names = []
+      @attached_policies_arns  = []
+      unless resp.empty?
+        resp.each do |r|
+          @attached_policies_names << r.policy_name
+          @attached_policies_arns  << r.policy_arn
+        end
+      end
+      return @attached_policies_names && @attached_policies_arns
```
Comment (r-fennell) [inline libraries/aws_iam_role.rb]: Do you need to return anything from this method? Can just add a bang `!` to the method and return nothing, as it updates `attached_policies_names` and `attached_policies_arns` in-place.

Comment (rmoles) [reply libraries/aws_iam_role.rb]: Ah meant to remove this return. Was re-writing this method for lazy loading, but changed my mind.

**Pull request merged**